networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/AttestationGossipManagerTest.java

```diff
@@ -52,39 +52,43 @@
   private final GossipEncoding gossipEncoding = GossipEncoding.SSZ_SNAPPY;
   private final TopicChannel topicChannel = mock(TopicChannel.class);
   private AttestationGossipManager attestationGossipManager;
+  private final AttestationSubnetSubscriptions attestationSubnetSubscriptions =
+      new AttestationSubnetSubscriptions(
+          gossipNetwork,
+          gossipEncoding,
+          attestationValidator,
+          recentChainData,
+          gossipedAttestationConsumer);
 
   @BeforeEach
   public void setup() {
     BeaconChainUtil.create(0, recentChainData).initializeStorage();
     doReturn(topicChannel).when(gossipNetwork).subscribe(contains("committee_index"), any());
-    AttestationSubnetSubscriptions attestationSubnetSubscriptions =
-        new AttestationSubnetSubscriptions(
-            gossipNetwork,
-            gossipEncoding,
-            attestationValidator,
-            recentChainData,
-            gossipedAttestationConsumer);
     attestationGossipManager =
         new AttestationGossipManager(gossipEncoding, attestationSubnetSubscriptions);
   }
 
   @Test
   public void onNewAttestation_afterMatchingAssignment() {
+    final Attestation attestation = dataStructureUtil.randomAttestation();
+    final Attestation attestation2 =
+        new Attestation(
+            dataStructureUtil.randomBitlist(),
+            dataStructureUtil.randomAttestationData(UnsignedLong.valueOf(13)),
+            dataStructureUtil.randomSignature());
+    final int subnetId = computeSubnetId(attestation);
+    // Sanity check the attestations are for the same subnet
+    assertThat(computeSubnetId(attestation2)).isEqualTo(subnetId);
```

Comment:
I'm confused with how these random attestations can have the same subnetId?

Reply:
Remember that we use a fixed seed for all randomness, so they're completely repeatable.  And then I carefully selected the slot for the second attestation to make it line up.